src/Eseye.php

```diff
@@ -323,53 +323,15 @@ public function invoke(string $method, string $endpoint, array $uri_data = []):
         }
 
         // Check if there is a cached response we can return
-        if (in_array(strtolower($method), $this->cachable_verb) &&
-            $cached = $this->getCache()->get($uri)
-        ) {
-            // In case the cached entry is still valid, mark content as being loaded from cache.
-            if (! $cached->expired())
-                $cached->setIsCachedLoad();
-
-            // Handling ETag marked response specifically (ignoring the expired time)
-            // Sending a request with the stored ETag in header - if we have a 304 response, data has not been altered.
-            if ($cached->hasHeader('ETag') && $cached->expired()) {
-
-                $result = $this->rawFetch($method, $uri, $this->getBody(), ['If-None-Match' => $cached->getHeader('ETag')]);
-
-                if ($result->getErrorCode() == 304) {
-
-                    // update expires header with newly provided value
-                    $cached->setExpires($result->expires());
-
-                    // store updated response in cache to renew internal cache duration
-                    $this->getCache()->set($uri, $cached);
-
-                    $cached->setIsCachedLoad();
-                }
-            }
-
-            // In case the result is effectively retrieved from cache,
-            // return the cached element.
-            if ($cached->isCachedLoad()) {
-
-                // Perform some debug logging
-                $this->getLogger()->debug('Loaded cached response for ' . $method . ' -> ' . $uri, [
-                    'etag' => $cached->hasHeader('ETag') ? $cached->getHeader('ETag') : '',
-                    'body' => $cached->raw,
-                ]);
-
-                $this->cleanupRequestData();
-
-                return $cached;
-            }
-        }
+        if ($this->isCachable($method))
+            $result = $this->handleCacheEntry($method, $uri);
```

Comment:
I assume this can be simplify to a direct access into cache handler since there will only have two issues :

- the endpoint can be cached
  - a cache entry exists
  - no cache entry exists
- the endpoint cannot be cached
  - no cache entry exists

```suggestion
$result = $this->handleCacheEntry($method, $uri);
```

Reply:
If there is a check when saving responses, this would make sense. The question is what's faster: checking if a string is in a small list of strings or going through a cache. Caches are made as an acceleration, but it still most likely involves IO operations, which make it inherently slow.

Reply:
Thing is there are more criteria than only the method which determine if yes or no an entry can be cached
And the handleCachedEntry is used to retrieve existing response based on a request
In case the response is not cachable, then we will of course not have it
However, in case the response is cachable, then, we might have it in cache
Last is to determine if yes or no the entry is still valid.

Reply:
I think the point about performance still holds true, even if we check a few other things. Maybe rename $method if it isn't actually the http method for clarity. Besides that, I think it doesn't really matter if we have the check or not.